Iso time

joesmolsky opened the pull request:
Change time format to ISO format.  This makes the TimeStuff class obsolete.  Times can now be set with datetime.utcnow().isoformat().

Version also changes from 1.1.0 to 1.1.1

KaraMelih reviewed (commented):
Looks quite good, thanks for taking care of it!
I just want to check it with pycharm to see if the TimeStuff is not used in anywhere else. 
This indeed simplifies things

KaraMelih commented:
Your commits from yesterday started to fail the tests;
https://github.com/SNEWS2/SNEWS_Publishing_Tools/actions
could you have a look at the cause

KaraMelih commented on snews_pt/snews_pub.py:
```diff
@@ -59,9 +60,15 @@ def send(self, messages):
+                self.display_message(message)
+            else:
+                click.secho(f'{"-" * 64}', fg='bright_red')
+                click.secho(f'Skipping message! Improper format! See log for details.')
```
Since it is only when checking the user input, isn't it easier to raise a warning rather than logging everything and directing user to the logs?

joesmolsky replied:
Yeah that sounds like a better way to do it.  I just did it this way because I already knew how to do logs and not warnings :wink:

KaraMelih commented on snews_pt/core/logging.py:
```diff
@@ -0,0 +1,56 @@
+HOST = gethostname()
+
+log_date = date.today().strftime("%Y-%m-%d")
+log_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../logs")
```
We create a directory relative to package. This works for the backend as you have to install the backend manually anyways. However, for the frontend, user likely to install the package via `pip install snews-pt` and it'll store the scripts somewhere in `.local/site-packages/...` whatever. Thus, the logs will also be created somewhere there which is not easy for user to find. 
My understanding is that this style logging is mostly for debugging and we should not expect that from the user

joesmolsky replied:
That makes sense.  I wasn't thinking about that.

joesmolsky commented:
I have updated the format checker to include warnings with the logs.

Also did some fixes to the time checking logic and fixed some typos in the test function times.

KaraMelih commented:
Thanks Joe it looks better!
I can try to test it tomorrow, latest next Monday and merge it. 
Please let me know if you need me to do it faster

KaraMelih commented on snews_pt/snews_pt_utils.py:
```diff
@@ -423,3 +427,110 @@ def get_name():
     return os.getenv("DETECTOR_NAME")
+
+
+def is_snews_format(snews_message, is_test=False):
```
`is_test` is never passed. Test scenarios are therefore not distinguished and can no longer be executed.
in cs I overcame this by checking for `this is a test` in the meta field 
[here](https://github.com/SNEWS2/SNEWS_Coincidence_System/blob/d69ce87ec52025f1f7a7668625a6e3f4c7ce6bea/snews_cs/cs_utils.py#L230)
and [here](https://github.com/SNEWS2/SNEWS_Publishing_Tools/blob/6c7d5bd0368c231cd13d5f8a884f7f9769277769/snews_pt/auxiliary/try_scenarios.py#L29) they are appended to every test message

KaraMelih replied:
instead of taking `is_test` as an argument (in which case we would need to pass around another `is_test` argument around within snews_pub) I'd say do a similar parsing within this function, and decide whether this is a test message or someone really sent a message with wrong input (e.g. neutrino times from future)

joesmolsky replied:
I changed the testing check to meta data like with snews-cs

KaraMelih commented on snews_pt/snews_pt_utils.py:
```diff
@@ -423,3 +427,110 @@ def get_name():
+    if missing_key:
+        warnings.warn(warning, UserWarning)
+        log.warning(warning)
+        return snews_format
```
if `missing_key` is True, it should return the `snews_format` but this can be referenced before assignment. e.g. there is detector_name but no neutrino time, so the `missing_key=True` but there is no `snews_format` defined. <br>
 In the beginning define a `snews_format=True` flag, and after the line 475 set it to False again maybe.

joesmolsky replied:
Done

KaraMelih reviewed (changes requested):
I made some comments. I noticed that there are a few minor things need to be changed

KaraMelih commented:
You can also debug these via `snews_pt run-scenarios` it shows already these error messages without having to interact with cs
